cns/service/main.go

```diff
@@ -795,6 +801,35 @@ func main() {
 		}
 	}
 
+	type CNSService struct {
+		pb.UnimplementedCNSServiceServer
+		Logger *zap.Logger
+	}
+
+	// Define gRPC server settings
+	settings := grpc.GrpcServerSettings{
+		IPAddress: "::",
+		Port:      8080,
+	}
+
+	// Initialize CNS service
+	cnsService := &CNSService{Logger: z}
+
+	// Create a new gRPC server
+	server, err := grpc.NewServer(settings, cnsService, z)
+	if err != nil {
+		logger.Errorf("Could not initialize gRPC server: %v", err)
+		os.Exit(1)
+	}
+
+	// Start the gRPC server
+	go func() {
+		if err := server.Start(); err != nil {
+			logger.Errorf("Could not start gRPC server: %v", err)
+			os.Exit(1)
+		}
+	}()
+
```

Comment:
Since all of this gRPC logic is for a new feature, let's put it all behind a feature flag. We should be able to still run CNS without executing any of the new code paths. Let's add a feature flag which can be toggled true or false through the CNS config.

Reply:
I have put the gRPC logic behind a feature flag by default I set it to false in the CNS config